### DISCUSSION TITLE
Force mysqldump to use different protocol than SOCKET

### DISCUSSION THREAD

**Discussion by duffysk** [dbeaver/dbeaver#11231] [Q&A] [answered]
Hello,

I am using DBeaver and I and I using dump feature, that calls mysqldump for MySQL database. Socket connection is not possible and I do not know how to force mysqldump to use -p=TCP instead of socket. Is it possible to add this feature?

**Comment by Maelstromeous** [comment]
This is a constant annoyance for me, this is the **only** thing that brings me back to MySQL Workbench, for being able to run import / export commands. Please add the ability to force mysqldump to use TCP sockets. If workbench can do it, dbbeaver can too!

It appears all that's needed is the `--protocol=tcp ` adding when the command is executed.

**Comment by Puxarnal** [comment, marked as answer]
Just add it as optional command parameter when dumping your database

![dbeaver_dump_protocol](https://user-images.githubusercontent.com/22577463/69797771-94784080-11d0-11ea-922d-c05df1ddc447.png)
